packages/mattermost-redux/src/reducers/entities/threads.ts

```diff
@@ -49,6 +49,14 @@ export const threadsReducer = (state: ThreadsState['threads'] = {}, action: Gene
     }
     case ThreadTypes.FOLLOW_CHANGED_THREAD: {
         const {id, following} = action.data;
+
+        // Delete local copy of thread when unfollowing
+        if (!following && state[id]) {
+            const nextState = {...state};
+            Reflect.deleteProperty(nextState, id);
+            return nextState;
+        }
```

Comment:
Without deleting the thread on unfollow the reply count and participants wouldn't update anymore locally since the thread selector was returning the unfollowed thread that was not being updated instead of creating a synthetic thread.

Reply:
// threadsReducer

// case PostTypes.RECEIVED_NEW_POST
// if newPost.root_id && state[newPost.root_id] && !state[newPost.root_id].is_following
// if ...!participants.find...
// updatedThread.participants = [...]

Reply:
I'd prefer to clear out stale data to handle it but let me see if it causes issues while viewing the thread in the global view. If it does then maybe it's easiest to do it in the threadsReducer

Reply:
> I'd prefer to clear out stale data to handle it but let me see if it causes issues while viewing the thread in the global view. If it does then maybe it's easiest to do it in the threadsReducer

Thinking about this again—if Global Threads (thread_item, thread_pane, etc.) switched to using `getThreadOrSynthetic` (instead `getThread`), it would probably work to clear stale data as you have it.

Reply:
Ok I'll check that

Reply:
@calebroseland turned out to be a lot more involved to use `getThreadOrSynthetic` and to keep the deletion of the thread on unfollow. Pushed a commit to instead go with your suggestion of adding the new post login to the threads reducer